addons/sale_timesheet/models/account.py

```diff
@@ -42,22 +42,20 @@ def _timesheet_preprocess(self, values):
         if 'task_id' in values:
             task = self.env['project.task'].sudo().browse(values['task_id'])
             values['so_line'] = task.sale_line_id.id or values.get('so_line', False)
+
+        # (re)compute the amount (depending on unit_amount, employee_id for the cost, and account_id for currency)
+        if values.get('project_id') and any([field_name in values for field_name in ['unit_amount', 'employee_id', 'account_id']]):
+            employee = self.env['hr.employee'].sudo().browse(values.get('employee_id'))
+            account = self.env['account.analytic.account'].sudo().browse(values.get('account_id'))
+            cost = employee.timesheet_cost or 0.0
```

Comment:
Since preprocess can also be called in `write`, you might not have `account_id` in values (case where you simply change the employee of existing timesheets). So I'd rather only deduce the uom in preprocess, based on the 'employee_id' values, and keep the postprocess as it is. This might causes many write of the same value, but you know, timesheet model is what it is .....

Reply:
Might be safer indeed :)